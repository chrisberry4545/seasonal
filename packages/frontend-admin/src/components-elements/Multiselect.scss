.c-multiselect {
  max-height: 200px;
  overflow-y: auto;
  border: 1px solid black;
}
