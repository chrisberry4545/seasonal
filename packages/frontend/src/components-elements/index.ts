export * from './BackgroundImageWithLoadingSpinner/BackgroundImageWithLoadingSpinner';

export * from './GroupedSelectBox/GroupedSelectBox';
