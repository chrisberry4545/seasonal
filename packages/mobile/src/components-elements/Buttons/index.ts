export * from './BareButton';

export * from './BorderedButton';
