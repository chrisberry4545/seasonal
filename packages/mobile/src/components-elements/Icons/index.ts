export * from './ArrowIcon';

export * from './BurgerIcon';

export * from './CrossIcon';

export * from './GridIcon';

export * from './ListIcon';

export * from './MagnifyingGlassIcon';
