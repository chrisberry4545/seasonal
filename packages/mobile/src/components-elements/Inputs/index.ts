export * from './Input';

export * from './MultilineInput';

export * from './RadioButton';

export * from './RadioButtonGroup';
