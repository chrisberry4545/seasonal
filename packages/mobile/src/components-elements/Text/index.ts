export * from './TextHeadingLarge';

export * from './TextHeadingMedium';

export * from './TextHeadingSmall';

export * from './TextLarge';

export * from './TextMedium';

export * from './TextSmall';
