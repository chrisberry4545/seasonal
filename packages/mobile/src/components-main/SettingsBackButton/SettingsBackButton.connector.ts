import { connect } from 'react-redux';
import {
  SettingsBackButton
} from './SettingsBackButton';
import {
  goBackFromSettingsPage
} from '@chrisb-dev/seasonal-shared-frontend-redux';

import {
  ISettingsBackButtonDispatchProps
} from './SettingsBackButton.interface';
import { Dispatch } from 'redux';

const mapDispatchToProps = (
  dispatch: Dispatch
): ISettingsBackButtonDispatchProps => ({
  onGoBack: () => dispatch(goBackFromSettingsPage())
});

export const SettingsBackButtonConnecter = connect(
  null,
  mapDispatchToProps
)(SettingsBackButton);
