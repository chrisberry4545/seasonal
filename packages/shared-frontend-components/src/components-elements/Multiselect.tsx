import React, { FC } from 'react';
import './Multiselect.scss';

export interface ISelectOption {
  label: string;
  value: string | number;
}

export const Multiselect: FC<{
  className?: string,
  onChange: (newValue: Array<(string | number)>) => void,
  options: ISelectOption[],
  value: Array<(string | number)>
}> = ({
  className,
  onChange,
  options,
  value
}) => {

  const isSelected = (optionValue: string | number) =>
    value.includes(optionValue);
  const onOptionClicked = (selectedValue: string | number) => {
    const newValue = isSelected(selectedValue)
      ? value.filter((innerValue) => innerValue !== selectedValue)
      : [...value, selectedValue];
    onChange(newValue);
  };

  return <div
    className={`c-multiselect ${(className || '')}`}>
      {
        options.map((option) =>
          <div key={option.value}
            className={isSelected(option.value) ? 'c-multi-select-option--selected' : ''}
            onClick={() => onOptionClicked(option.value)}>
              {option.label}
          </div>
        )
      }
  </div>;
};
