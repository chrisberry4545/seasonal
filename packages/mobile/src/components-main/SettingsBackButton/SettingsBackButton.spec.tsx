import React from 'react';
import { SettingsBackButton } from './SettingsBackButton';
import { shallow, ShallowWrapper } from 'enzyme';
import { BackArrowIconButton } from '../../components-elements';

jest.mock('../../components-elements', () => ({
  BackArrowIconButton: () => 'BackArrowIconButton'
}));

describe('<SettingsBackButton />', () => {
  let wrapper: ShallowWrapper;
  let mockOnGoBack: jest.Mock;

  beforeEach(() => {
    mockOnGoBack = jest.fn();
    wrapper = shallow(
      <SettingsBackButton
        onGoBack={mockOnGoBack} />
    );
  });

  test('renders correctly', () => expect(wrapper).toMatchSnapshot());

  test('can click the go back button', () => {
    const onClick = wrapper.find(BackArrowIconButton).first().props().onClick;
    if (onClick) {
      onClick();
    }
    expect(mockOnGoBack).toHaveBeenCalled();
  });

});
