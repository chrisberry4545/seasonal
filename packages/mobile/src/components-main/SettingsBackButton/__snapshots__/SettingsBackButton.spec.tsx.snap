// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SettingsBackButton /> renders correctly 1`] = `
<BackArrowIconButton
  onClick={[MockFunction]}
  style={
    Object {
      "marginRight": 12,
      "marginTop": 1,
    }
  }
/>
`;
